### PR TITLE
Compile button and option to disable automatic compilation

### LINKED ISSUE #101: Add Feature: Compile Only On Demand
Currently, browser-solidity recompiles every time a change has been made to the file. This may be problematic, however, if the user is working with very large files as the whole browser freezes up to compile every time a change is made to a large project.

There should be an option to only compile on demand.

### PR DESCRIPTION
**Pull request by axic**
Fixes  #101

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -436,6 +436,41 @@ var run = function () {
 
   var renderer = new Renderer(editor, executionContext.web3(), updateFiles, udapp, executionContext, formalVerification.event, compiler.event); // eslint-disable-line
 
+  var autoCompile = document.querySelector('#autoCompile').checked;
+
+  document.querySelector('#autoCompile').addEventListener('change', function () {
```
Comment (chriseth) [inline src/app.js]: We have jquery here, so you could do some live stuff, but it's of course also fine like that.

File: src/app.js [hunk 2]
```diff
@@ -436,6 +436,41 @@ var run = function () {
...
+  var compileTimeout = null;
+
+  function editorOnChange () {
+    if (!autoCompile) {
+      return;
+    }
+
+    var input = editor.getValue();
+    if (input === '') {
+      editor.setCacheFileContent('');
```
Comment (chriseth) [inline src/app.js]: Don't we want to do this even if we do not compile?

File: src/app.js (hunk 1 above)
Comment (axic) [inline src/app.js]: This matches the rest of the file. I think it should be updated at a later stage to only use jQuery.

File: src/app.js (hunk 2 above)
Comment (axic) [inline src/app.js]: I don't fully know what is the point of that code, but moved the check past it.

**Pull request merged**